python/setup.py

```diff
@@ -31,6 +31,17 @@ def get_build_type():
         # TODO: change to release when stable enough
         return "TritonRelBuildWithAsserts"
 
+
+def get_codegen_backends():
+    backends = []
+    env_prefix = "TRITON_CODEGEN_"
+    for name, _ in os.environ.items():
+        if name.startswith(env_prefix) and check_env_flag(name):
+            assert name.count(env_prefix) <= 1
+            backends.append(name.removeprefix(env_prefix))
+    return backends
```

Comment:
Build different backends according to the environment variables that start with `TRITON_CODEGEN_`. Take the Intel XPU backend as an example. If the user sets `TRITON_CODEGEN_INTEL_XPU_BACKEND=1`, the `INTEL_XPU_BACKEND` will be added to the codegen backends and then pass it to cmake to build it.